[iOS] Fixed auto picker resizing after selection

kubaflo opened the pull request:
### Description of Change

Fixed auto picker resizing after selection

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/19295


https://github.com/dotnet/maui/assets/42434498/f790fb6a-f5a1-4412-8ef3-08370c1081d2

rmarinho reviewed (changes requested):
Hi @kubaflo thanks for your contribution, is it possible to add a UITest for this? 

thanks.

Hottemax commented on src/Core/src/Platform/iOS/PickerExtensions.cs:
```diff
@@ -61,6 +62,11 @@ internal static void UpdatePicker(this MauiPicker platformPicker, IPicker picker
 
 			pickerView?.Select(Math.Max(selectedIndex, 0), 0, true);
+
+			if (shouldInvalidateMessure)
+			{
+				picker.InvalidateMeasure();
+			}
```
Messure -> Measure (typo)
But since it is initialized 20 lines above its first use, it is even easier to understand and maintain when simply inlining the condition

```suggestion
			if (newSelectedIndex != picker.SelectedIndex)
			{
				picker.InvalidateMeasure();
			}
```

kubaflo replied:
This cannot be the case because on the line 57 we assign selectedIndex (newSelectedIndex if not null)

Hottemax commented on src/Core/src/Platform/iOS/PickerExtensions.cs:
```diff
@@ -41,6 +41,7 @@ internal static void UpdateAttributedPlaceholder(this MauiPicker platformPicker,
 		internal static void UpdatePicker(this MauiPicker platformPicker, IPicker picker, int? newSelectedIndex = null)
 		{
 			var selectedIndex = newSelectedIndex ?? picker.SelectedIndex;
+			bool shouldInvalidateMessure = newSelectedIndex != picker.SelectedIndex;
```
```suggestion
```

kubaflo replied:
Thanks for catching it

kubaflo commented:
@rmarinho I added a UiTest

jsuarezruiz commented:
/azp run

ScottAtBrady commented:
I need this fix to release my software.  Can we get this merged and put into a release of Maui?

jsuarezruiz commented:
/azp run

jsuarezruiz commented:
/azp run

kubaflo committed:
[iOS] Fixed auto picker resizing after selection

Update pr-20207.md

Improve iOS picker resize test and add helper method

Refactored the Issue19295 test to use constants and a new SelectPickerWheelValue helper for selecting picker items on iOS. Updated UITest configuration to use fixed device name and platform version for iOS tests. Added SelectPickerWheelValue extension method to simplify picker selection in Appium-based tests.

Create pr-20207.md
